### ISSUE TITLE
Ability to open files/folders with a terminal-based editors (e.g. Helix, Vim, Emacs)

### ISSUE THREAD

**Issue by massung** [warpdotdev/Warp#2320]
### Describe the solution you'd like?

```text
Right now in the settings the only editors usable are separate apps (e.g. VSCode). It would be great to be able to choose editors like Helix, Vim, Emacs, etc. and be able to optionally open the file/folder in a new Warp tab (or split pane).
```

### How important is this feature to you?

3

### Warp Internal (ignore) - linear-label:770f6576-d6c0-4e4f-a259-fc64b5156087

None

**Comment by dannyneira** [comment]
Thanks for this feature request! Please :+1:  and subscribe to this request as it helps us gauge interest. We will post any updates here.

Just to clarify you want to have a way to open files/folders in terminal text editors with a mouse ( like with vscodes "Explorer" )?

**Comment by massung** [comment]
Right. For example, if I cmd-click a filename it opens VSCode automatically with the file. This is great!

But - personally - I prefer Helix and being able to choose that as my editor of choice so that when I cmd-click a file it launches Helix either: inline (same tab, same pane), in a new tab, or in a new split pane within the same tab, would be fantastic.

One might ask "if you're right there, why not just type `hx <filename>`?"

A reasonable question, too. At least for myself, a common use-case would be something like:

```
find . -name '*.csv' | grep ...
```

So the file in question might be in some arbitrarily deep path. Having the option to cmd-click the output to open the file is already so helpful. This would just be icing. But I recognize it might be a low priority item.

**Comment by dannyneira** [comment]
I get what you mean, thanks for clarifying! As a possible stopgap for this, check out [lunarvim](https://www.lunarvim.org/), it has some of these features built into the Text-based IDE

**Comment by 7flash** [comment]
I am using Helix with Warp, and really enjoy it, usually I have multiple tabs for different projects in Helix opened in separate tabs within Warp terminal. Currently, I can rename tabs and then navigate between them with Cmd+1,2

Another feature I would like to request that would be awesome if tab title would automatically show currently open file or at least directory. Thanks @dannyneira

**Comment by jonasgheer** [comment]
> I am using Helix with Warp, and really enjoy it, usually I have multiple tabs for different projects in Helix opened in separate tabs within Warp terminal. Currently, I can rename tabs and then navigate between them with Cmd+1,2
> 
> Another feature I would like to request that would be awesome if tab title would automatically show currently open file or at least directory. Thanks @dannyneira

How did you avoid Warp swallowing inputs? For instance cmd+o (expand select) does not work if I launch Helix from Warp, but it does work in iTerm2.

**Comment by babucarr32** [comment]
Any updates on this?
